example/concurrency-01.rst

```diff
@@ -0,0 +1,22 @@
+fn loop_and_print() {
+    let x = 0;
+    loop {
+        if x == 10 {
+            break;
+        }
+        x += 1;
+        println(x);
+    }
+}
+
+println("Creating threads");
+
+let thread_id_1 = spawn loop_and_print;
```

Comment:
This is good, but is it ok if we do it like in Go where it's a function call instead so we can support passing args too? e.g

```
fn hi(x: int) {
     println(x);
}

let x : int = 20;
let t = spawn hi(x);
join t;
```

So spawn expects a function call (I can enforce this even at the parse level), and arguments are evaluated in the outer environment and then passed into the function where the body can run concurrently. If the function returns a function then we would have to do brackets again like
```
spawn hof(x,y)();
```